### DISCUSSION TITLE
Error while build & install cheroot & CherryPy from sources

### DISCUSSION THREAD

**Discussion by gabriel-yahav** [cherrypy/cheroot#23] [Q&A] [answered]
Hi, 

I'm trying to build & intall CherryPy 10.2.1 from sources . For doing that, I've already installed 
all the dependency python modules needed successfully. 

However, when i tried finally to build the cheroot and cherrypy modules I got the following error : 

"error in CherryPy setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers."

"error in cheroot setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers."
 
cheroot sources  : cheroot-5.3.0.tar.gz
CherryPy sources : cherrypy-10.2.1.tar.gz
Python version : 2.7.5

I'll be glad for any help , 
Thanks !

**Comment by jaraco** [comment, marked as answer]
You need a later version of Setuptools that supports environment markers in requires directives.

**Comment by webknjaz** [comment]
@jaraco can we add minimum setuptools version to `setup_requires`?

**Comment by jaraco** [comment]
Unfortunately, [no](https://github.com/pypa/setuptools/issues/1002).

**Comment by webknjaz** [comment]
Shouldn't we close this issue then?

**Comment by jaraco** [comment]
Yes, sure.
